File: DataLayer/models/address.py
class Address:
    __jobAdvertVacantJobId = int
    __streetAddress = str
    __city = str
    __country = str
    __postalCode = str

    def __init__(self, job_advert_vacant_job_id: int, street_address: str, city: str, country: str, postal_code: str):
        """
        Creates an object of Address.
        Args:
            job_advert_vacant_job_id: Unique object id, for the jobadvert object entity.
            street_address: The address of the job advert post.
            city: The city of the job advert post.
            country: The country of the job advert post.
            postal_code: The postal code for the city.
        """
        self.__jobAdvertVacantJobId = job_advert_vacant_job_id
        self.__streetAddress = street_address
        self.__city = city
        self.__country = country
        self.__postalCode = postal_code

    @property
    def id(self) -> int:
        return self.__jobAdvertVacantJobId

    @property
    def street_address(self) -> str:
        return self.__streetAddress

    @property
    def city(self) -> str:
        return self.__city

    @property
    def country(self) -> str:
        return self.__country

    @property
    def postal_code(self) -> str:
        return self.__postalCode

File: ObjectLibrary/Common/Address.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ObjectLibrary.Common
{
    /// <summary>
    /// Handles the information of an address.
    /// </summary>
    public class Address : BaseEntity
    {
        private string _streetAddress;
        private string _city;
        private string _country;
        private string _postalCode;

        public Address(int jobAdvertVacantJobId, string streetAddress, string city, string country, string postalCode) : base (jobAdvertVacantJobId)
        {
            _streetAddress = streetAddress;
            _city = city;
            _country = country;
            _postalCode = postalCode;
        }

        /// <summary>
        /// Contains the specific Street Address.
        /// </summary>
        public string StreetAddress { get { return _streetAddress; } set { _streetAddress = value; } }

        /// <summary>
        /// The city of which the address is located.
        /// </summary>
        public string City { get { return _city; } set { _city = value; } }

        /// <summary>
        /// The country of which the city is in.
        /// </summary>
        public string Country { get { return _country; } set { _country = value; } }

        /// <summary>
        /// The postal code of which the city is associated to.
        /// </summary>
        public string PostalCode { get { return _postalCode; } set { _postalCode = value; } }
    }
}
